events/config.go

```diff
@@ -0,0 +1,74 @@
+package events
+
+import (
+	"errors"
+	"github.com/Shopify/sarama"
+	"github.com/kelseyhightower/envconfig"
+)
+
+const DeadLetterSuffix = "-dead-letters"
+
+type CloudEventsConfig struct {
+	EventSource           string   `envconfig:"CLOUD_EVENTS_SOURCE" required:"true"`
+	KafkaBrokers          []string `envconfig:"KAFKA_BROKERS" required:"true"`
+	KafkaConsumerGroup    string   `envconfig:"KAFKA_CONSUMER_GROUP" required:"false"`
+	KafkaTopic            string   `envconfig:"KAFKA_TOPIC" default:"events"`
+	KafkaDeadLettersTopic string   `envconfig:"KAFKA_DEAD_LETTERS_TOPIC"`
+	KafkaTopicPrefix      string   `envconfig:"KAFKA_TOPIC_PREFIX" required:"true"`
+	KafkaRequireSSL       bool     `envconfig:"KAFKA_REQUIRE_SSL" required:"true"`
+	KafkaVersion          string   `envconfig:"KAFKA_VERSION" required:"true"`
+	KafkaUsername         string   `envconfig:"KAFKA_USERNAME" required:"false"`
+	KafkaPassword         string   `envconfig:"KAFKA_PASSWORD" required:"false"`
+	SaramaConfig          *sarama.Config
+}
+
+func NewConfig() *CloudEventsConfig {
+	cfg := &CloudEventsConfig{}
+	cfg.SaramaConfig = sarama.NewConfig()
+	cfg.SaramaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	return cfg
+}
+
+func (k *CloudEventsConfig) LoadFromEnv() error {
+	if err := envconfig.Process("", k); err != nil {
+		return err
+	}
+	version, err := sarama.ParseKafkaVersion(k.KafkaVersion)
+	if err != nil {
+		return err
+	}
+	k.SaramaConfig.Version = version
+	if k.KafkaRequireSSL {
+		k.SaramaConfig.Net.TLS.Enable = true
+	}
+	if k.KafkaUsername != "" && k.KafkaPassword != "" {
+		k.SaramaConfig.Net.SASL.Enable = true
+		k.SaramaConfig.Net.SASL.Mechanism = sarama.SASLTypeSCRAMSHA512
+		k.SaramaConfig.Net.SASL.User = k.KafkaUsername
+		k.SaramaConfig.Net.SASL.Password = k.KafkaPassword
+	}
+
+	return nil
+}
+
+func (k *CloudEventsConfig) GetPrefixedTopic() string {
+	return k.KafkaTopicPrefix + k.KafkaTopic
+}
+
+func (k *CloudEventsConfig) GetDeadLettersTopic() string {
+	if k.KafkaDeadLettersTopic == "" {
+		return k.KafkaDeadLettersTopic
+	}
+	return k.KafkaTopicPrefix + k.KafkaDeadLettersTopic
+}
+
+func (k *CloudEventsConfig) IsDeadLettersEnabled() bool {
+	return k.GetDeadLettersTopic() != ""
```

Comment:
Should you overload the meaning in this way?  Would you want an empty string to mean the default dead letter topic?

Reply:
This might cause issues when multiple services are using the default - we could default to the `EventSource`. We should discuss this later today.

Reply:
Given that we'll have a dead letter topic per topic and consumer group pair, I prefer if this setting is set explicitly.